tests/functional/metadataIngestion/ingestion.js

```diff
@@ -0,0 +1,185 @@
+    { path: '/queue-populator/logState/raft_2/logOffset', value: '1' },
+    { path: '/queue-populator/logState/raft_3/logOffset', value: '1' },
+    { path: '/queue-populator/logState/raft_4/logOffset', value: '1' },
+    { path: '/queue-populator/logState/raft_5/logOffset', value: '1' },
+    { path: '/queue-populator/logState/raft_6/logOffset', value: '1' },
+    { path: '/queue-populator/logState/raft_7/logOffset', value: '1' },
+    { path: '/queue-populator/logState/raft_8/logOffset', value: '1' },
+];
+
+describe.only('Ingest metadata to kafka', () => {
+    let metadataMock;
+    let httpServer;
+    let kafkaConsumer;
+    let queuePopulator;
+    let zkClient;
+
+    before(function before(done) {
+        async.waterfall([
+            next => {
+                kafkaConsumer = new kafka.KafkaConsumer(testKafkaConfig);
+                return next();
+            },
+            next => {
+                kafkaConsumer.connect({ allTopics: true }, (err, res) => {
+                    console.log('attempting to connect to kafkaConsumer');
+                    console.log(err, res);
+                    return next(err);
+                });
+            },
+            next => {
+                metadataMock = new MetadataMock();
+                httpServer = http.createServer((req, res) =>
+                    metadataMock.onRequest(req, res)).listen(7779);
+                return next();
+            },
+            // next => {
+            //     iProducer = new IngestionProducer({
+            //         host: 'localhost:7779',
+            //         port: 7779,
+            //     });
+            //     return next();
+            // },
+            next => {
+                zkClient = zookeeper.createClient('127.0.0.1:2181');
+                zkClient.connect();
+                zkClient.once('error', (err, res) => {
+                    console.log('error connecting to zookeeper');
+                    console.log(err, res);
+                    throw err;
+                });
+                zkClient.once('ready', (err, res) => {
+                    console.log('zkclient is ready');
+                    console.log(err, res);
+                    return next();
+                });
+            },
+            next => {
+                return async.each(testZkPaths, (path, cb) => {
+                    return zkClient.mkdirp(path, (err, res) => {
+                        console.log('trying to mkdirp', err, res);
+                        return cb(err, res);
+                    });
+                }, next);
+            },
+            next => {
+                return async.each(logOffsetPaths, (logOffset, cb) => {
+                    const buf = Buffer.from(logOffset.value, 'utf-8');
+                    console.log('here is the buffer', buf);
+                    return async.waterfall([
+                        next => zkClient.mkdirp(logOffset.path, buf, (err, res) => {
```

Comment:
you should pass `null` as `buf` on the mkdirp, as intermediate nodes should not have data attached.